engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/CreateDeploymentTest.java

```diff
@@ -642,6 +642,29 @@ public void shouldRejectDeploymentWithInvalidTimerStartEventExpression() {
                 + "'INVALID_CYCLE_EXPRESSION')\n");
   }
 
+  @Test
+  public void shouldRejectDeploymentWhenNoMessageReferenced() {
+    // given
+    final BpmnModelInstance definition =
+        Bpmn.createExecutableProcess("processId")
+            .startEvent("startEvent")
+            .messageEventDefinition()
+            .id("messageEvent")
+            .done();
+
+    // when
+    final Record<DeploymentRecordValue> deployment =
+        ENGINE.deployment().withXmlResource("process.bpmn", definition).expectRejection().deploy();
+
+    // then
+    Assertions.assertThat(deployment)
+        .hasRejectionType(RejectionType.INVALID_ARGUMENT)
+        .hasRejectionReason(
+            "Expected to deploy new resources, but encountered the following errors:\n"
+                + "'process.bpmn': - Element: messageEvent\n"
+                + "    - ERROR: Must reference a message\n");
+  }
+
```

Comment:
:x: First of all, the test case is nicely written :+1: Since the class contains already many test cases, we should add the new test case somewhere else.

The unit test `ProcessMessageStartEventMessageNameValidatorTest` would be an option. The downside is that it just test the validator itself.

I suggest creating a new test class for message validation tests. It could look similar to `TimerValidationTest` (without the parameterized style) and follow the "new" JUnit 5 style.

In order to verify the expected message using `ProcessValidationUtil`. We need to add the `ZeebeDesignTimeValidators` to the validation result.